pyccel/ast/cwrapper.py

```diff
@@ -378,10 +378,18 @@ def C_to_Python(c_object):
     -------
     FunctionDef : cast type FunctionDef
     """
-    try :
-        cast_function = c_to_py_registry[(c_object.dtype, c_object.precision)]
-    except KeyError:
-        errors.report(PYCCEL_RESTRICTION_TODO, symbol=c_object.dtype,severity='fatal')
+    if c_object.rank != 0:
+        if c_object.order == 'C':
+            cast_function = 'c_ndarray_to_pyarray'
+        elif c_object.order == 'F':
+            cast_function = 'fortran_ndarray_to_pyarray'
+        else:
+            cast_function = 'ndarray_to_pyarray'
```

Comment:
Shouldn't we distinguish between C and Fortran ordering only when `c_object.rank > 1`?

Reply:
In theory `c_object.order` should be equal to `None` if `c_object.rank == 1`. The function `ndarray_to_pyarray` adds the C and Fortran flags if the rank is 1. See [here](https://github.com/pyccel/pyccel/blob/ebourne_expose_array/pyccel/stdlib/cwrapper_ndarrays/cwrapper_ndarrays.c#L226-L232)